solr/core/src/java/org/apache/solr/response/BinaryResponseWriter.java

```diff
@@ -196,14 +197,16 @@ public static NamedList<Object> getParsedResponse(SolrQueryRequest req, SolrQuer
       }
       Resolver resolver = new Resolver(req, rsp.getReturnFields());
 
-      ByteArrayOutputStream out = new ByteArrayOutputStream();
-      try (JavaBinCodec jbc = new JavaBinCodec(resolver)) {
-        jbc.setWritableDocFields(resolver).marshal(rsp.getValues(), out);
-      }
+      try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+        try (JavaBinCodec jbc = new JavaBinCodec(resolver)) {
+          jbc.setWritableDocFields(resolver).marshal(rsp.getValues(), out);
+        }
 
-      InputStream in = out.toInputStream();
```

Comment:
again, consider BAOS to avoid the extra byte array & copy

Reply:
Resolved in c6cf531 by @dsmiley